mmhuman3d/data/data_structures/human_data.py

```diff
@@ -241,6 +260,90 @@ def dump(self, npz_path: str, overwrite: bool = True):
+                                slice_end = min((slice_index + 1) * slice_size,
+                                                self.__temporal_len__)
+                                slice_list[sub_dim] = \
+                                    slice(slice_start, slice_end)
+                                sliced_sub_value = sub_value[tuple(slice_list)]
+                                if key not in sliced_data[str(slice_index)]:
+                                    sliced_data[str(slice_index)][key] = {}
+                                sliced_data[str(slice_index)][key][sub_key] = \
+                                    sliced_sub_value
+                        else:
+                            for slice_index in range(slice_num):
+                                slice_start = slice_index * slice_size
+                                slice_end = min((slice_index + 1) * slice_size,
+                                                self.__temporal_len__)
+                                sliced_sub_value = \
+                                    sub_value[slice(slice_start, slice_end)]
+                                if key not in sliced_data[str(slice_index)]:
+                                    sliced_data[str(slice_index)][key] = {}
+                                sliced_data[str(slice_index)][key][sub_key] = \
+                                    sliced_sub_value
+                if len(non_sliced_sub_dict) > 0:
+                    non_sliced_data[key] = non_sliced_sub_dict
+            else:
+                value = self.get_raw_value(key)
+                # slice as ndarray
+                if isinstance(value, np.ndarray):
+                    slice_list = [
+                        slice(None),
+                    ] * len(value.shape)
+                    for slice_index in range(slice_num):
+                        slice_start = slice_index * slice_size
+                        slice_end = min((slice_index + 1) * slice_size,
+                                        self.__temporal_len__)
+                        slice_list[dim] = slice(slice_start, slice_end)
+                        sliced_value = value[tuple(slice_list)]
+                        sliced_data[str(slice_index)][key] = sliced_value
+                # slice as list/tuple
+                else:
+                    for slice_index in range(slice_num):
+                        slice_start = slice_index * slice_size
+                        slice_end = min((slice_index + 1) * slice_size,
+                                        self.__temporal_len__)
+                        sliced_value = value[slice(slice_start, slice_end)]
+                        sliced_data[str(slice_index)][key] = sliced_value
+        writer_args_dict = {
+            'slice_size': slice_size,
+            'keypoints_info': keypoints_info,
+            'non_sliced_data': non_sliced_data,
+            'key_strict': self.get_key_strict()
+        }
+        return writer_args_dict, sliced_data
+
```

Comment:
When generating cache for one npz, it will double the RAM at most, and both HumanData and HumanDataCacheWriter will be freed after cache writing. Thus the peak comes to 2*(RAM of the biggest HumanData).
Fortunately, we are releasing compressed HumanData to avoid zero-padding, and OS always helps Python swap useless memory. So far there's nothing to concern about if we generate cache on a server, with a single process.